### PR TITLE
[-] Cart Rule : Improved Customer Back Office display

### PR DESCRIPTION
**Pull request by mickaelandrieu**
- [x] display all discount values and methods
- [X] if the global quantity equals 0, don't display
- [X] if the available quantity for the customer equals 0, don't display
- [X] display tax information for minimum amount

### CODE REVIEW

**Comment by xBorderie** [comment]
Could you please amend your commit names to fit the guidelines? Thank you!

**Comment by mickaelandrieu** [comment]
@xBorderie done :) I tagged it as "FO" but should be "Customer BO" I'm not realy sure :/

**Comment by Shudrum** [comment]
Ping @AlexEven ! :bath:

**Comment by xBorderie** [comment]
And ping @vTerenti ! :)

File: themes/default-bootstrap/discount.tpl [hunk 1]
```diff
@@ -48,21 +48,27 @@
...
+						{/if}
+						{if $discountDetail.free_shipping}
+							{if $discountDetail.reduction_percent > 0 || $discountDetail.reduction_amount > 0} + {/if}
 							{l s='Free shipping'}
-						{else}
-							-
+						{/if}
+						{if $discountDetail.gift_product > 0}
+							{if $discountDetail.reduction_percent > 0 || $discountDetail.reduction_amount > 0 || $discountDetail.gift_product} + {/if}
+							{$discountDetail.gift_product_name} {l s='free'}!
```
Comment (Quetzacoalt91) [inline themes/default-bootstrap/discount.tpl]: If in a language, the word "free" must appear before the product name, we can use this syntax:
```php
{l s='Free %s !' sprintf=$discountDetail.gift_product_name}
```

Comment (AlexEven) [inline themes/default-bootstrap/discount.tpl]: Even better: {l s='Free %s!' sprintf=$discountDetail.gift_product_name}
(no space before exclamation mark in English)

**Comment by vTerenti** [comment]
QA OK

**Comment by xBorderie** [comment]
@mickaelandrieu Is it a work in progress, or is it testable? :)

**Comment by mickaelandrieu** [comment]
@xBorderie  it's finished since months (24/12) vTerenti only asked me a very little update.

**Comment by xBorderie** [comment]
Cool, thanks! :)

File: controllers/front/DiscountController.php [hunk 2]
```diff
@@ -39,15 +39,26 @@ public function initContent()
     {
         parent::initContent();
 
-        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, false);
+        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, true, true);
```
Comment (brrrrrrrt) [inline controllers/front/DiscountController.php]: with this change you are setting "includeGeneric" to true, which means that all Vouchers will be displayed not just the personal ones, are you really sure that this is what you want?
imho, making all Vouchers visible to all customers is a major change in the way vouchers work.

Comment (brrrrrrrt) [inline controllers/front/DiscountController.php]: just upgraded a shop to 1.6.1.6 and noticed **all Vouchers can be viewed by the customers** :(

#5797 thx! helped though.

Comment (yannishimself) [inline controllers/front/DiscountController.php]: Is there any insight about why this takes place now? How should secret vouchers be distributed now? Had to roll this back in my case.

Comment (xBorderie) [inline controllers/front/DiscountController.php]: @yanngarras This issue was fixed in #5978, which is included [in v1.6.1.7](http://build.prestashop.com/news/prestashop-1617-maintenance-release/). Please upgrade any 1.6.1.6 shop to 1.6.1.7!

**Pull request merged**